mtags/src/main/scala/scala/meta/internal/pc/CompletionProvider.scala

```diff
@@ -378,18 +378,18 @@ class CompletionProvider(
       val isTypeMember = kind == CompletionListKind.Type
       params.checkCanceled()
       val matchingResults = completions.matchingResults { entered => name =>
+        val decoded = entered.decoded
         /** NOTE(tgodzik): presentation compiler bug https://github.com/scala/scala/pull/8193
          *  should be removed once we drop support for 2.12.8 and 2.13.0
          *  in case we have a comment presentation compiler will see it as the name
          *  CompletionIssueSuite.issue-813 for more details
          */
         val realEntered =
-          if (entered.startsWith("$div$div") ||
-            entered.startsWith("$div$times")) { // start of a comment: "//" or "/*"
+          if (decoded.startsWith("//") || decoded.startsWith("/*")) { // start of a comment
             // we reverse the situation and look for the word from start of complition to either '.' or ' '
-            val reversedString = entered.toString().reverse
-            val lastSpace = reversedString.indexOfSlice("0200u$") // reversed $u0020 = ' '
-            val lastDot = reversedString.indexOfSlice("E200u$") // reversed $u002E = '.'
+            val reversedString = decoded.reverse
+            val lastSpace = reversedString.indexOfSlice(" ")
+            val lastDot = reversedString.indexOfSlice(".")
             val startOfWord =
               if (lastSpace < lastDot && lastSpace >= 0) lastSpace else lastDot
             reversedString.slice(0, startOfWord).reverse
```

Comment:
The only thing I wasn't 100% sure on is here do we need to also encode this again before we pass it into `matchesSubCharacters` or `matches` since the `entered.toString`?

Reply:
Not sure either, but might be good to encode it again. Situations like: `weird_method@Name`

Reply:
We always want to fuzzy match against the decoded name, it would be weird to include `::` if the user types `foo.$colon$col@@`